h2/connection.py

```diff
@@ -586,8 +593,11 @@ def initiate_upgrade_connection(self, settings_header=None):
             For servers, returns nothing.
         :rtype: ``bytes`` or ``None``
         """
-        frame_data = None
+        self.config.logger.debug(
+            "Upgrade connection.\nSettings frame: %s", self.local_settings
```

Comment:
> As a further note: should we consider structlog?

I think that should be a user decision but that does make this quite difficult to do consistently at that point. Not every user will want to use structlog and those that do just want the "default" logging will want this. If there were a way to simply handle both, I'd be in favor of that (and I've been looking for a way of discerning that for other libraries).